packages/@react-stately/layout/src/TableLayout.ts

```diff
@@ -155,10 +157,8 @@ export class TableLayout<T> extends ListLayout<T> {
       // or the content of the item changed.
       let previousLayoutNode = this.layoutNodes.get(node.key);
       if (previousLayoutNode) {
-        let curNode = this.collection.getItem(node.key);
-        let lastNode = this.lastCollection ? this.lastCollection.getItem(node.key) : null;
         height = previousLayoutNode.layoutInfo.rect.height;
-        isEstimated = curNode !== lastNode || width !== previousLayoutNode.layoutInfo.rect.width || previousLayoutNode.layoutInfo.estimatedSize;
+        isEstimated = node !== previousLayoutNode.node || width !== previousLayoutNode.layoutInfo.rect.width || previousLayoutNode.layoutInfo.estimatedSize;
```

Comment:
I put a console log here when trying to debug why the selection checkbox was still using its estimated height and noticed that it was being infinitely called with isEstimated always returning true for each checkbox cell/row header